packages/mui-material/src/Autocomplete/Autocomplete.js

```diff
@@ -1081,7 +1081,7 @@ Autocomplete.propTypes /* remove-proptypes */ = {
    *
    * @param {React.SyntheticEvent} event The event source of the callback.
    * @param {Value} option The highlighted option.
-   * @param {string} reason Can be: `"keyboard"`, `"auto"`, `"mouse"`, `"touch"`.
+   * @param {string} reason Can be: `"keyboard"`, `"mouse"`, `"touch"`.
```

Comment:
Why is `"auto"` being removed? Under which scenarios is `"auto"` the reason?

Reply:
The `auto` reason is the default value for `setHighlightedIndex` under which `onHighlightChange` is triggered. It was introduced in #20691 [here](https://github.com/mui/material-ui/pull/20691/files#diff-2a1a679552b21721859eeece5ca3014abc3d94cfb72ef81c241142798a223831R124) and is used when no `keyboard`, `mouse`, or `touch` interaction occurs, like maybe in the cases noted in the bug report and PR description. With the updated logic, `reason` should now be limited to `keyboard`, `mouse`, or `touch` interactions.